### PR TITLE
Adds chugging, knocking on a table and slamming your hand on a table.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
- Added chugging. Clickdrag a reagent container you're holding in your hand to your character.
  - Added knocking and slamming your hand on a table. While aiming for right or left hand, grab intent click a table to knock, harm intent to slam.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

**Comment by atteria** [comment]
Grab intent plus mouth targeting sounds kinda obscure and easy to do unintentionally.
Another input I've seen for chugging other servers is dragging a glass that's in your hand on your sprite. I'd suggest that.

**Comment by NonQueueingMatt** [comment]
> Grab intent plus mouth targeting sounds kinda obscure and easy to do unintentionally. Another input I've seen for chugging other servers is dragging a glass that's in your hand on your sprite. I'd suggest that.

done

**Review by Alberyk — APPROVED** [review]

**Review by alsoandanswer — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1756,15 +1756,24 @@
 		return 1
 	return 0
 
+/mob/living/carbon/human/proc/can_drink(var/obj/item/I)
+	if(!check_has_mouth())
+		to_chat(src, "Where do you intend to put \the [I]? You don't have a mouth!")
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:1761]: no span intended?

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1756,15 +1756,24 @@
...
+		return FALSE
+	var/obj/item/blocked = check_mouth_coverage()
+	if(blocked)
+		to_chat(src, "<span class='warning'>\The [blocked] is in the way!</span>")
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human.dm:1765]: span macro?

File: code/modules/reagents/reagent_containers.dm [hunk 3]
```diff
@@ -283,6 +278,46 @@
...
+			return
+
+		if(!H.can_drink())
+			return
+
+		if(!(H.l_hand == src) && !(H.r_hand == src))
+			return
+
+		if(!reagents.total_volume)
+			to_chat(H, "<span class='notice'>\The [src] is empty.</span>")
```
Comment (SleepyGemmy) [inline code/modules/reagents/reagent_containers.dm:295]: no span macro?

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**